### PR TITLE
[enocean] Add metadata channels for rssi, repeatcount and last received timestamp

### LINKED ISSUE #4591: [enocean] Add metadata channels
I think of splitting the channel "receivingState" into its parts RSSI & repeater count to log these values in database, to see anomalies. I think the best would be to remove "receivingState", because one can build this string on his own. Furthermore, I'd like to add a channel for the time when the last telegram was received.

See https://github.com/fruggy83/openocean/issues/37

### PR DESCRIPTION
**Pull request by dominikkv**
The three channels are `advanced="true"`, so they do not appear in channel overview. The old channel `receivingState` is deleted.

Signed-off-by: Dominik Krickl-Vorreiter <dominikkv@gmx.de>
closes #4591
See fruggy83/openocean#56

### CODE REVIEW

**Comment by dominikkv** [comment]
Hi @fruggy83, thanks for your feedback. I have pulled almost all readme changes from your repo, not only the ones regarding the metadata channels. There were also some changes from you, so I have added the Also-By Tag in the commit message. I hope this is OK.

**Comment by dominikkv** [comment]
I've also updated documentation regarding channel configuration like requested in #4585

**Comment by dominikkv** [comment]
Did a rebase to catch up with master, hoping to resolve some weird Jenkins errors, but the Jenkins build still fails. Any suggestions?

**Comment by wborn** [comment]
Looks like all builds currently fail with the same exception:

```
Caused by: java.io.IOException: Could not create temp dir /tmp/tycho1635332495760495036equinox
	at org.eclipse.sisu.equinox.embedder.internal.DefaultEquinoxEmbedder.copyToTempFolder(DefaultEquinoxEmbedder.java:192)
	at org.eclipse.sisu.equinox.embedder.internal.DefaultEquinoxEmbedder.doStart(DefaultEquinoxEmbedder.java:139)
	at org.eclipse.sisu.equinox.embedder.internal.DefaultEquinoxEmbedder.start(DefaultEquinoxEmbedder.java:67)
	at org.eclipse.sisu.equinox.embedder.internal.DefaultEquinoxEmbedder.checkStarted(DefaultEquinoxEmbedder.java:309)
	... 47 more
```

Looks like some storage/permission issue that can perhaps be resolved by cleaning/restarting the machine @kaikreuzer ?

**Comment by wborn** [comment]
The weird Jenkins errors are gone now @dominikkv. :smile:

**Comment by fruggy83** [comment]
Sorry for the delay, had to stay in hospital for the last three weeks. Thanks a lot for your work @dominikkv

File: addons/binding/org.openhab.binding.enocean/README.md [hunk 1]
```diff
@@ -202,15 +218,42 @@ The channels of a thing are determined automatically based on the chosen EEP.
 | dimmer              | Dimmer             | Dimmer value in percent |
 | generalSwitch(/A/B) | Switch             | Switch something (channel A/B) ON/OFF |
 | rollershutter       | Rollershutter      | Shut time (shutTime) in seconds can be configured |
+| angle               | Number:Angle       | The angle for blinds
```
[the remark this answers is not in the document]
Comment (dominikkv) [reply addons/binding/org.openhab.binding.enocean/README.md]: Thanks! Done.

File: addons/binding/org.openhab.binding.enocean/README.md [hunk 2]
```diff
@@ -202,15 +218,42 @@ The channels of a thing are determined automatically based on the chosen EEP.
...
+|energyStorage        | Number:ElectricPotential | Energy storage, don't know what this means...|
+|rssi                 | Number                   | Received Signal Strength Indication (dBm) of last received message |
+|repeatCount          | Number                   | Number of repeaters involved in the transmission of the telegram |
+|lastReceived         | DateTime                 | Date and time the last telegram was received |
+
+Items linked to bi-directional actuators (actuator sends status messages back) should always disable the ```autoupdate```.
+This is especially true for Eltako rollershutter, as their position is calcaulted out of the current position and the moving time.
+
+## Channel Configuration
+Some channels can be configured with parameters.
```
[the remark this answers is not in the document]
Comment (dominikkv) [reply addons/binding/org.openhab.binding.enocean/README.md]: Jup, done. Thanks.

File: addons/binding/org.openhab.binding.enocean/README.md [hunk 3]
```diff
@@ -56,20 +69,21 @@ Hence if your device supports one of the following EEPs the chances are good tha
...
 | automatedMeterSensor            | A5-12       | 0x00-03       | counter, currentNumber, instantpower,<br/>totalusage, amrLitre, amrCubicMetre | FWZ12 | Discovery |
 | centralCommand                  | A5-38       | 0x08          | dimmer, generalSwitch        | Eltako FUD14, FSR14            | Teach-in |
 | rollershutter                   | A5-3F/D2-05/A5-38 | 0x7F/00/08 | rollershutter             | Eltako FSB14, NodOn SIN-2-RS-01| Teach-in/Discovery |
 | measurementSwitch               | D2-01       | 0x00-0F,11,12 | generalSwitch(/A/B), instantpower,<br/>totalusage, repeaterMode | NodOn In Wall Switch | Discovery |
 | classicDevice                   | F6-02       | 0x01-02       | virtualRockerswitchA, virtualRockerswitchB | - | Teach-in |
 
-¹ Not all channels are supported by all devices, it depends which specific EEP type is used by the device, all thing types additionally support receivingState channel
+¹ Not all channels are supported by all devices, it depends which specific EEP type is used by the device, all thing types additionally support rssi, repeatCount and lastReceived channels
```
[the remark this answers is not in the document]
Comment (dominikkv) [reply addons/binding/org.openhab.binding.enocean/README.md]: Surrounded each name with single back ticks.

File: addons/binding/org.openhab.binding.enocean/README.md [hunk 4]
```diff
@@ -202,15 +218,42 @@ The channels of a thing are determined automatically based on the chosen EEP.
...
 | teachInCMD          | Switch             | Sends a teach-in msg, content can configured with parameter teachInMSG |
 | virtualSwitchA      | Switch             | Used to convert switch item commands into rocker switch messages (channel A used)<br/>Time in ms between sending a pressed and release message can be defined with channel parameter duration.<br/>The switch mode (rocker switch: use DIR1 and DIR2, toggle: use just one DIR) can be set with channel parameter switchMode (rockerSwitch, toggleButtonDir1, toggleButtonDir2)|
 |virtualRollershutterA| Rollershutter      | Used to convert rollershutter item commands into rocker switch messages (channel A used)|
 |rockerswitchListenerSwitch| Switch        | Used to convert rocker switch messages into switch item state updates|
 |rockerswitchListenerRollershutter| Rollershutter | Used to convert rocker switch messages into rollershutter item state updates|
 |virtualRockerswitchB | String             | Used to send plain rocker switch messages (channel B used)|
+|batteryVoltage       | Number:ElectricPotential | Battery voltage for things with battery|
```
[the remark this answers is not in the document]
Comment (dominikkv) [reply addons/binding/org.openhab.binding.enocean/README.md]: Added whitespaces in this table.

File: addons/binding/org.openhab.binding.enocean/README.md [hunk 5]
```diff
@@ -202,15 +218,42 @@ The channels of a thing are determined automatically based on the chosen EEP.
...
+|energyStorage        | Number:ElectricPotential | Energy storage, don't know what this means...|
+|rssi                 | Number                   | Received Signal Strength Indication (dBm) of last received message |
+|repeatCount          | Number                   | Number of repeaters involved in the transmission of the telegram |
+|lastReceived         | DateTime                 | Date and time the last telegram was received |
+
+Items linked to bi-directional actuators (actuator sends status messages back) should always disable the ```autoupdate```.
```
[the remark this answers is not in the document]
Comment (dominikkv) [reply addons/binding/org.openhab.binding.enocean/README.md]: Removed unnecessary backticks.

File: addons/binding/org.openhab.binding.enocean/README.md [hunk 6]
```diff
@@ -202,15 +218,42 @@ The channels of a thing are determined automatically based on the chosen EEP.
...
+This is especially true for Eltako rollershutter, as their position is calcaulted out of the current position and the moving time.
```
[the remark this answers is not in the document]
Comment (dominikkv) [reply addons/binding/org.openhab.binding.enocean/README.md]: Fixed spelling.

File: addons/binding/org.openhab.binding.enocean/README.md [hunk 7]
```diff
@@ -202,15 +218,42 @@ The channels of a thing are determined automatically based on the chosen EEP.
...
+
+## Channel Configuration
+Some channels can be configured with parameters.
+
+|Channel type    | Parameter      | Meaning                    | Possible Values |
```
[the remark this answers is not in the document]
Comment (dominikkv) [reply addons/binding/org.openhab.binding.enocean/README.md]: Thanks, I have rebuilt the table with your suggested table formatter.

File: addons/binding/org.openhab.binding.enocean/README.md [hunk 8]
```diff
@@ -202,15 +218,42 @@ The channels of a thing are determined automatically based on the chosen EEP.
...
+|----------------|----------------|----------------------------|---|
+| rollershutter  | shutTime       | Time (in seconds) to completely close the rollershutter |  |
+| dimmer         | rampingTime    | Duration of dimming | A5-38-08: Ramping Time (in seconds), 0 = default ramping, 1..255 = seconds to 100%; D2-01-01: 0 = switch, 1-3 = timer 1-3, 4 = stop |
+| teachInCMD     | manufacturerId | Id is used for 4BS teach in with EEP   | HEX |
+|                | teachInMSG     | Use this message if teach in type and/or manufacturer id are unknown | HEX |
+
+Possible declaration in Thing DSL:
```
[the remark this answers is not in the document]
Comment (dominikkv) [reply addons/binding/org.openhab.binding.enocean/README.md]: Done, thanks!

File: addons/binding/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/eep/EEP.java [hunk 9]
```diff
@@ -96,8 +98,21 @@ public State convertToState(String channelId, String channelTypeId, Configuratio
...
 
-        if (channelTypeId.equals(CHANNEL_RECEIVINGSTATE)) {
-            return convertToReceivingState();
+        switch (channelTypeId) {
+            case CHANNEL_RSSI:
+                if (this.optionalData == null || this.optionalData.length < 6) {
+                    return UnDefType.UNDEF;
+                }
+
+                return new QuantityType<>((this.optionalData[5] & 0xFF) * -1, SmartHomeUnits.ONE);
```
[the remark this answers is not in the document]
Comment (dominikkv) [reply addons/binding/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/eep/EEP.java]: That is simplier, thanks!

**Comment by dominikkv** [comment]
Thank you very much, @cweitkamp, for your code review. I hope I have adressed all of your suggested changes. The documentation is reformated, State types are changed, and a PR for adding a hint for the user about channel removal is opened. 

The DCO says that I haven't signed my work, but as far as I can see there is a Signed-off-by in every commit. Furthermore, Jenkins is not done yet, there is a decent queue and each build takes more than an hour...

Please let me know if I can do any more.

**Comment by dominikkv** [comment]
Thanks! My name has changed recently, that's why the older commits are signed with Krickl-Vorreiter. The right name now is Vorreiter.
I have changed the name in the git settings from "dominikkv" to "Dominik Vorreiter", maybe this will help DCO in the future.

**Pull request merged**